test/parallel/test-net-socket-connect-without-cb.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+require('../common');
+const net = require('net');
+
+const server = net.createServer((conn) => {
+  conn.end();
+  server.close();
+}).listen(0, () => {
+  const client = new net.Socket();
+
+  client.on('connect', () => {
```

Comment:
I am on it. but then `test/parallel/test-net-socket-connecting.js` also need to be changed in future?

anyway for this PR, i will only wrap `test-net-socket-connect-without-cb.js` callback with `common.mustCall`